torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -162,7 +162,10 @@ std::shared_ptr<FutureMessage> pyRpcPythonUdf(
       std::vector<char>(pickledPythonUDF.begin(), pickledPythonUDF.end()),
       tensors);
   return sendMessageWithAutograd(
-      agent, dst, std::move(*pythonCall).toMessage());
+      agent,
+      dst,
+      std::move(*pythonCall).toMessage(),
+      true /*forceGradRecording*/);
```

Comment:
Adding `forceGradRecording` because UDFs might not carry any requires_grad tensor, but does requires grad on the return value.

Reply:
IIUC, this fixes https://github.com/pytorch/pytorch/issues/28819. If so, can we have a simple unit test for the problem this fixes?

Reply:
Yes, actually let me split this PR into two, with the first one focusing on fixing #28819 with unit test.